lib/ansible/modules/system/group.py

```diff
@@ -120,9 +120,14 @@ def group_del(self):
         cmd = [self.module.get_bin_path(command_name, True), self.name]
         return self.execute_command(cmd)
 
+    def _local_check_gid_exists(self):
+        if self.gid and self.gid in [gr.gr_gid for gr in grp.getgrall()]:
+            self.module.fail_json(msg="GID '{0}' already exists".format(self.gid))
```

Comment:
do we want this message only on 'add' or also on mod?

Reply:
The check method is called in both `group_add` and `group_mod`, see below. Or are you saying we shouldn't do the check in `mod`?